gaps/gap-25_golem_certificates/gap-25_golem_certificates.md

```diff
@@ -0,0 +1,191 @@
+
+The required properties are:
+- `subject.displayName` the name to be displayed when presenting the certificate to a user
+- `subject.contact.email` an email address where the subject can be reached for any kind inquiry
+
+##### Public Key
+
+The `publicKey` property stores the public key of the subject's key pair to allow verification of their signatures. The node defines the following properties:
+- `publicKey.algorithm` the encryption algorithm for which the key pair is generated for
+- `publicKey.parameters` any parameters that are relevant to key or the algorithm for proper usage
+- `publicKey.key` the hexadecimal encoded string representation of the public key optionally with a "0x" prefix
+
+##### Validity Period
+
+The `validityPeriod` property defines the time interval for which the certificate is valid. It is defined by the start of the term in `validityPeriod.notBefore` property and the end of the term in the `validityPeriod.notAfter` property.
+
+##### Key usage
+
+The `keyUsage` property defines how the key attached to the certificate can be used. We define the following usages:
+- `signCertificate` the key can be used to sign other certificates to extend the chain n trust
+- `signManifest` the key can be used to sign payload manifests (defined in GAP-5) /manifest will not support Golem certificates right now, it requires some update to GAP-5 and the implementation/
+- `signNode` the key can be used to sign a node descriptor (defined in GAP-XX)
+- `all` the key can be used without restrictions, including use cases that are defined after the certificate was created
+
+##### Permissions
+
+Details of how the holder of the certificate is permitted to use the Golem network. The content of this property is governed by the Permissions schema.
+##### Signature
+
+This object contains the details of the signature validating the subject and providing integrity protection for properties in the `certificate` object. It defines the following properties:
+- `algorithm` the cryptographic algorithm that was used to create the signature, it requires two details:
+  - `hash` the hash function used to create the fingerprint of the signed data
+  - `encryption` the cryptographic algorithm used to encrypt the fingerprint
+- `signer` for certificates the signer can the string `self` meaning that the certificate was self signed (aka Root certificate) ot an object containing a signed certificate. We opted to include the full certificate as it is easier to handle and understand when opening the JSON certificate in a text editor. We do not anticipate long certificate chains where this would a problem and we think that requiring the certificate to be accessible via the internet would limit the usage for individuals.
```

Comment:
```suggestion
- `signer` for certificates the signer can the string `self` meaning that the certificate was self signed (aka Root certificate) or an object containing a signed certificate. We opted to include the full certificate as it is easier to handle and understand when opening the JSON certificate in a text editor. We do not anticipate long certificate chains where this would a problem and we think that requiring the certificate to be accessible via the internet would limit the usage for individuals.
```